deebot_client/commands/xml/pos.py

```diff
@@ -16,7 +16,25 @@
     from deebot_client.event_bus import EventBus
 
 
-class GetPos(XmlCommandWithMessageHandling):
+class PosParser:
+    """Support class for producing Pos events."""
+
+    @classmethod
+    def _parse_xml(
+        cls, position_type: PositionType, event_bus: EventBus, xml: Element
+    ) -> HandlingResult:
+        """Handle xml message and notify the correct event subscribers."""
+        if (p := xml.attrib.get("p")) and (xml.attrib.get("valid", "1")) == "1":
```

Comment:
```suggestion
        if (p := xml.attrib.get("p")) and xml.attrib.get("valid") == "1":
```

We currently have tests, where always valid is present. Therefore we should analyse the message, when its missing instead assuming it's valid

Reply:
valid=1 is missing from GetChargerPos though, I can either move the check to GetPos and Pos and remove it from the parser or assume it's "1" if the attribute is missing